bindings/pydeck/pydeck/bindings/deck.py

```diff
@@ -76,20 +84,26 @@ def __init__(
         self.views = views
         # Use passed view state
         self.initial_view_state = initial_view_state
-        self.deck_widget = DeckGLWidget()
-        self.deck_widget.custom_libraries = pydeck_settings.custom_libraries
-        api_keys = api_keys or {}
-        self._set_api_keys(api_keys)
 
-        self.deck_widget.height = height
-        self.deck_widget.width = width
-        self.deck_widget.tooltip = tooltip
+        api_keys = api_keys or {}
 
         self.description = description
         self.effects = effects
-
         self.map_provider = str(map_provider).lower() if map_provider else None
-        self.deck_widget.map_provider = map_provider
+        self._tooltip = tooltip
+
+        if has_jupyter_extra():
+            from ..widget import DeckGLWidget
```

Comment:
As noted above, instead of re-importing again and again, I think it would be clearer (and maybe faster?) to try to import `DeckGLWidget` at the top level, and otherwise set it to `None`

```py
try:
    from ..widget import DeckGLWidget
except ImportError:
    DeckGLWidget = None
```

Then just check
```py
if DeckGLWidget:
    self.deck_widget = DeckGLWidget()
    ...
```